azure/services/virtualmachines/client.go

```diff
@@ -78,15 +78,29 @@ func (ac *AzureClient) CreateOrUpdateAsync(ctx context.Context, spec azure.Resou
 	ctx, _, done := tele.StartSpanWithLogger(ctx, "virtualmachines.AzureClient.CreateOrUpdate")
 	defer done()
 
-	vm, err := ac.vmParams(ctx, spec)
+	var existingVM interface{}
+
+	if existing, err := ac.Get(ctx, spec.ResourceGroupName(), spec.ResourceName()); err != nil && !azure.ResourceNotFound(err) {
+		return nil, nil, errors.Wrapf(err, "failed to get virtual machine %s in %s", spec.ResourceName(), spec.ResourceGroupName())
```

Comment:
shouldn't we handle errors that are not `ResourceNotFound`?

Reply:
if it's a `ResourceNotFound` error, that means the VM doesn't exist, so we don't set existingVM. It's not an error we want to return, it's an expected error whenever we create the resource for the first time.

Reply:
oh yeah, this used to be in a different form in `virtualmachines.go`. Perhaps I should be more in sync with the new code, but alas the code itself is not in sync anymore.